tests/functional/execute_child_pipeline/test_execute_child_pipeline_activity.py

```diff
@@ -4,10 +4,13 @@
 from data_factory_testing_framework.state import RunParameter, RunParameterType
 
 
-def test_execute_pipeline_activity_child_activities_executed(request: pytest.FixtureRequest) -> None:
+@pytest.mark.parametrize("framework_type", [TestFrameworkType.DataFactory, TestFrameworkType.Synapse])
+def test_execute_pipeline_activity_child_activities_executed(
```

Comment:
I see we took a bit of a strange direction of including both `fabric` and `data-factory` tests for functional tests that are actually framework-type agnostic!  (e.g. switch_activity and execute_child_pipeline functional tests) In essence those tests provide no real value and merely introduce test code duplication. Let's have a call to see how we can maybe address this as part of your PR, because you built on top of these tests?

Reply:
Summary of our discussion:
1. Functional tests that are FrameworkType agnostic should not require to be tested for multiple frameworks (i.e. `execute_child_activity` and `switch_activity` tests). Remove the `pytest.mark.parametrize` from these tests and hardcode the framework_type for the corresponding pipeline definition alongside the test.
2. Introduce unit / functional tests that cover the Synapse and Fabric portion of the code (i.e. `evaluate_activities`, `repositories` and `test_framework`)
3. Optional - as you mentioned code coverage, have a look at something like the following extension: https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters

Reply:
If code coverage can be added would be great to add that extension to the dev container 👍

Reply:
I agree @LeonardHd. I will tackle code coverage in a separate PR though.